save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/benchmarks/TabMenuBar.kt

```diff
@@ -0,0 +1,58 @@
+package com.saveourtool.save.entities.benchmarks
+
+import com.saveourtool.save.domain.Role
+
+/**
+ * Interface for tab bar in many pages
+ */
+interface TabMenuBar<T> {
+    /**
+     * Default value in every Enum classes
+     */
+    val defaultTab: T
+
+    /**
+     * Regular expression for url classification
+     */
+    val regexForUrlClassification: Regex
+
+    /**
+     * Pair consisting of a short address of the default tab and a long prefix of the addresses of all other tabs
+     */
+    var paths: Pair<String, String>
+
+    /**
+     * @return Array of elements this Enum
+     */
+    fun values(): Array<T>
+
+    /**
+     * @param elem
+     * @return The Enum element by the string it corresponds to, or an exception if it is not found
+     */
+    fun valueOf(elem: String): T
+
+    /**
+     * @param elem
+     * @return Equivalent to valueOf(), but returns null instead of an exception
+     */
+    fun findEnumElement(elem: String): T?
```

Comment:
```suggestion
    fun findEnumElement(elem: String): T? = values().firstOrNull { valueOf(elem) == it }
```
Maybe like this we can add default implementation for this method